app/src/main/java/org/mifos/mobile/ui/fragments/LoanApplicationFragment.java

```diff
@@ -547,6 +547,8 @@ public void onItemSelected(AdapterView<?> parent, View view, int position, long
 
             case R.id.sp_loan_purpose:
                 if (loanTemplate.getLoanPurposeOptions() != null) {
+                    if (position == loanTemplate.getLoanPurposeOptions().size())
+                        position--;
```

Comment:
can you explain why are we doing this?

Reply:
Actually this line of code was regarding to another issue that has been raised by me. During committing I have written the wrong issue number in the commit message. I did update the commit message but I don't know why the change didn't appear here.

Reply:
I removed this change. I find out it has already been done by another developer.